File: NOVEMBER-2025/23-11-2025/1262.go
func maxSumDivThree(nums []int) int {
    sum := 0

    m1a, m1b := int(^uint(0)>>1), int(^uint(0)>>1)
    m2a, m2b := int(^uint(0)>>1), int(^uint(0)>>1)

    for _, x := range nums {
        sum += x
        r := x % 3

        if r == 1 {
            if x < m1a {
                m1b = m1a
                m1a = x
            } else if x < m1b {
                m1b = x
            }
        } else if r == 2 {
            if x < m2a {
                m2b = m2a
                m2a = x
            } else if x < m2b {
                m2b = x
            }
        }
    }

    rem := sum % 3
    if rem == 0 {
        return sum
    }

    remove := int(^uint(0) >> 1)

    if rem == 1 {
        opt1 := m1a
        opt2 := int(^uint(0) >> 1)
        if m2b < int(^uint(0)>>1) {
            opt2 = m2a + m2b
        }
        if opt1 < opt2 {
            remove = opt1
        } else {
            remove = opt2
        }
    } else {
        opt1 := m2a
        opt2 := int(^uint(0) >> 1)
        if m1b < int(^uint(0)>>1) {
            opt2 = m1a + m1b
        }
        if opt1 < opt2 {
            remove = opt1
        } else {
            remove = opt2
        }
    }

    if remove == int(^uint(0)>>1) {
        return 0
    }
    return sum - remove
}

File: NOVEMBER-2025/23-11-2025/1262.kt
class Solution {
    fun maxSumDivThree(nums: IntArray): Int {
        var sum = 0

        var m1a = Int.MAX_VALUE
        var m1b = Int.MAX_VALUE
        var m2a = Int.MAX_VALUE
        var m2b = Int.MAX_VALUE

        for (x in nums) {
            sum += x
            val r = x % 3

            if (r == 1) {
                if (x < m1a) {
                    m1b = m1a
                    m1a = x
                } else if (x < m1b) {
                    m1b = x
                }
            } else if (r == 2) {
                if (x < m2a) {
                    m2b = m2a
                    m2a = x
                } else if (x < m2b) {
                    m2b = x
                }
            }
        }

        val rem = sum % 3
        if (rem == 0) return sum

        var remove = Int.MAX_VALUE

        if (rem == 1) {
            val opt1 = m1a
            val opt2 = if (m2b < Int.MAX_VALUE) m2a + m2b else Int.MAX_VALUE
            remove = minOf(opt1, opt2)
        } else {
            val opt1 = m2a
            val opt2 = if (m1b < Int.MAX_VALUE) m1a + m1b else Int.MAX_VALUE
            remove = minOf(opt1, opt2)
        }

        return if (remove == Int.MAX_VALUE) 0 else sum - remove
    }
}